src/features/FindReferencesManager.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2019 - present Adobe. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var AppInit                     = require("utils/AppInit"),
+        CommandManager              = require("command/CommandManager"),
+        Commands                    = require("command/Commands"),
+        EditorManager               = require("editor/EditorManager"),
+        ProviderRegistrationHandler = require("features/PriorityBasedRegistration").RegistrationHandler,
+        SearchResultsView           = require("search/SearchResultsView").SearchResultsView,
+        SearchModel                 = require("search/SearchModel").SearchModel,
+        Strings                     = require("strings");
+
+    var _providerRegistrationHandler = new ProviderRegistrationHandler(),
+        registerFindReferencesProvider = _providerRegistrationHandler.registerProvider.bind(
+            _providerRegistrationHandler
+        ),
+        removeFindReferencesProvider = _providerRegistrationHandler.removeProvider.bind(_providerRegistrationHandler);
+
+    var searchModel = new SearchModel(),
+        _resultsView;
+
+    function _getReferences(provider, hostEditor, pos) {
+        var result = new $.Deferred();
+
+        if(!provider) {
```

Comment:
Since you are using _resultsView and AppInit takes some time to dispatch AppReady (this should be **htmlReady** right? as in [FindinFilesUI.js](https://github.com/adobe/brackets/blob/master/src/search/FindInFilesUI.js#L454)), you should put a null check for _resultsView here as well.